File: src/test/java/org/veri/be/unit/auth/JwtClaimsPayloadTest.java
package org.veri.be.unit.auth;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;
import org.veri.be.domain.member.entity.Member;
import org.veri.be.domain.member.entity.enums.ProviderType;
import org.veri.be.global.auth.JwtClaimsPayload;

class JwtClaimsPayloadTest {

    @Nested
    @DisplayName("from")
    class From {

        @Test
        @DisplayName("멤버 정보를 클레임으로 변환한다")
        void mapsMemberToClaims() {
            Member member = Member.builder()
                    .id(1L)
                    .email("member@test.com")
                    .nickname("member")
                    .profileImageUrl("https://example.com/profile.png")
                    .providerId("provider-1")
                    .providerType(ProviderType.KAKAO)
                    .build();

            JwtClaimsPayload payload = JwtClaimsPayload.from(member);

            assertThat(payload.id()).isEqualTo(1L);
            assertThat(payload.email()).isEqualTo("member@test.com");
            assertThat(payload.nickName()).isEqualTo("member");
            assertThat(payload.isAdmin()).isFalse();
        }
    }
}

File: tests/src/test/kotlin/org/veri/be/unit/auth/JwtClaimsPayloadTest.kt
package org.veri.be.unit.auth

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import org.veri.be.global.auth.JwtClaimsPayload

class JwtClaimsPayloadTest {

    @Nested
    @DisplayName("of")
    inner class Of {

        @Test
        @DisplayName("클레임 정보를 생성한다")
        fun mapsFieldsToClaims() {
            val payload = JwtClaimsPayload.of(1L, "member@test.com", "member", false)

            assertThat(payload.id()).isEqualTo(1L)
            assertThat(payload.email()).isEqualTo("member@test.com")
            assertThat(payload.nickName()).isEqualTo("member")
            assertThat(payload.isAdmin()).isFalse()
        }
    }
}
